doc/GUIDE.md

```diff
@@ -0,0 +1,848 @@
+and passed back to jextract:
+
+```sh
+$ jextract --dump-includes includes.txt mylib.h
+$ grep Foo includes.txt > includes_filtered.txt
+$ jextract @includes_filtered.txt mylib.h
+```
+
+Users should exercise caution when filtering symbols, as it is relatively easy to filter
+out a declaration that is depended on by one or more declarations:
+
+```c
+// test.h
+struct A {
+   int x;
+}
+struct A aVar;
+```
+
+Here, we could run jextract and filter out `A`, like so:
+
+```sh
+$ jextract --include-var aVar test.h
+```
+
+However, doing so would lead to broken generated code, as the layout of the global variable
+`aVar` depends on the layout of the excluded struct `A`.
+
+In such cases, jextract will report the missing dependency and terminate without
+generating any bindings:
+
+```txt
+ERROR: aVar depends on A which has been excluded
+```
+
+## Tracing
+
+It is sometimes useful to inspect the parameters passed to a native call, especially when
+diagnosing application bugs and/or crashes. The code generated by the jextract tool
+supports _tracing_ of native calls, that is, parameters passed to native calls can be
+printed on the standard output.
+
+To enable the tracing support, just pass the `-Djextract.trace.downcalls=true` flag to the
+launcher used to start the application. Below we show an excerpt of the output when
+running the [OpenGL example](samples/opengl) with tracing support enabled:
+
+```txt
+glutInit(MemorySegment{ address: 0x7fa6b03d6400, byteSize: 4 }, MemorySegment{ address: 0x7fa6b03d6400, byteSize: 4 })
+glutInitDisplayMode(18)
+glutInitWindowSize(900, 900)
+glutCreateWindow(MemorySegment{ address: 0x7fa6b03f8e70, byteSize: 14 })
+glClearColor(0.0, 0.0, 0.0, 0.0)
+glShadeModel(7425)
+glLightfv(16384, 4611, MemorySegment{ address: 0x7fa6b03de8d0, byteSize: 16 })
+glLightfv(16384, 4608, MemorySegment{ address: 0x7fa6b0634840, byteSize: 16 })
+glLightfv(16384, 4609, MemorySegment{ address: 0x7fa6b0634840, byteSize: 16 })
+glLightfv(16384, 4610, MemorySegment{ address: 0x7fa6b0634840, byteSize: 16 })
+glMaterialfv(1028, 5633, MemorySegment{ address: 0x7fa6b0634860, byteSize: 4 })
+glEnable(2896)
+glEnable(16384)
+glEnable(2929)
+glutDisplayFunc(MemorySegment{ address: 0x7fa6a002e820, byteSize: 0 })
+glutIdleFunc(MemorySegment{ address: 0x7fa6a015a620, byteSize: 0 })
+glutMainLoop()
+glClear(16640)
+glPushMatrix()
+glRotatef(-20.0, 1.0, 1.0, 0.0)
+glRotatef(0.0, 0.0, 1.0, 0.0)
+glutSolidTeapot(0.5)
+```
+
+## Preprocessor Definitions
+
+C header files are processed by a pre-processor by a compiler before they are inspected
+further. It is possible for a header file to contain so-called 'compiler switches', which
+can be used to conditionally generate code based on the value of a macro, for instance:
+
+```c
+#ifdef MY_MACRO
+int x = 42;
+#else
+int x = 0;
+#endif
+```
+
+The value of these macros also affects the behavior of jextract. Therefore, jextract
+supports setting macro values on the command line using the `-D` or
```

Comment:
Perhaps mentioning the distinctions between these macros and the ones described in the Constants section more specifically will be helpful?

Reply:
Hmm, not sure what to say. There are no real distinctions, but jextract only generates an accessor if a macro is defined in the header files, not if it is only set on the command line with `-D`. (I'll add that at least)